### PR TITLE
issue/TAPC-16 done

### PR DESCRIPTION
**Pull request by EmiliyanPanagyurski**

### CODE REVIEW

File: week3/emo/addUser.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from authentication.authenticate import authenticate
+
+@authenticate(passwordLength=12)
+def addUser(user):
+   """ This method puts user in database """
+   connection.set(user['username'], user)
```
Comment (rbagrov) [inline week3/emo/addUser.py]: Isn't this what you've ment?

```python
connection.set(user['username'], user['password'])
```

File: week3/emo/authentication/authenticate.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+from db.connection import connection
+
+def authenticate(passwordLength=8):
+    """ This method authenticates the user before he is put into the database """
+    def outer_wrapper(func):
+        def wrapper(user, *args, **kwargs):
+            if len(user['password']) > passwordLength:
+                print('Authenticating user')
+                userExists = connection.get(user['username'])
+                if userExists == 'nil':
```
Comment (rbagrov) [inline week3/emo/authentication/authenticate.py]: When ```userExists``` can be string and this string can have value of ```nil``` ?

File: week3/emo/authentication/authenticate.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                    print('access granted')
+                    func(user, *args, **kwargs)
+                else:
+                    print('User is already in the database')
```
Comment (rbagrov) [inline week3/emo/authentication/authenticate.py]: You are actually not storing even valid users.

File: week3/emo/authentication/authenticate.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+            else:
+                print('password must be longer than ' + str(passwordLength))
```
Comment (rbagrov) [inline week3/emo/authentication/authenticate.py]: Please use ```.format```

File: week3/emo/db/connection.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# This file contains db configuration
+import redis
+
+connection = redis.StrictRedis(host='localhost', port=6379, db=0)
```
Comment (rbagrov) [inline week3/emo/db/connection.py]:
```
HOST = 'localhost'
PORT = 6379
DB = 0

connection = redis.StrictRedis(host=HOST, port=PORT, db=DB)

File: week3/emo/db/connection.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# This file contains db configuration
+import redis
```
Comment (rbagrov) [inline week3/emo/db/connection.py:2]: Where do you install ```redis``` ?

**Review by rbagrov — CHANGES_REQUESTED** [review]

File: week3/emo/authentication/authenticate.py (hunk 2 above)
Comment (EmiliyanPanagyurski) [reply week3/emo/authentication/authenticate.py]: Yes actually it can!

File: week3/emo/db/connection.py (hunk 6 above)
Comment (EmiliyanPanagyurski) [reply week3/emo/db/connection.py:2]: I have a question about the requirements in chat.you might want to check it.

File: week3/emo/authentication/authenticate.py (hunk 2 above)
Comment (rbagrov) [reply week3/emo/authentication/authenticate.py]: Please check here: https://github.com/andymccurdy/redis-py/blob/master/redis/client.py#L973-L977
I suspect your intentions are to check if response object is None, which is correct.
In python there is no nil and it is not string. Python equivalent is None and it has its own type.

File: week3/emo/addUser.py (hunk 1 above)
Comment (EmiliyanPanagyurski) [reply week3/emo/addUser.py]: No, the key corresponds to a dict!And the key is the username.

File: week3/emo/authentication/authenticate.py (hunk 3 above)
Comment (EmiliyanPanagyurski) [reply week3/emo/authentication/authenticate.py]: How come? Please elaborate!

Comment (rbagrov) [reply week3/emo/authentication/authenticate.py]: I don't remember when I checked that it has the client class. Anyways now setting works.

File: week3/emo/authentication/authenticate.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+def authenticate(passwordLength=8):
+    """ This method authenticates the user before he is put into the database """
+    def outer_wrapper(func):
+        def wrapper(user, *args, **kwargs):
+            logger.info('checking to so if {} is valid'.format(user['username']))
+            if len(user['password']) > passwordLength:
+                logger.info('Authenticating user')
+                userExists = client.getItem(user['username'])
+                if userExists == None:
```
Comment (rbagrov) [inline week3/emo/authentication/authenticate.py]: better representations would be: ```if not userExists:```

File: week3/emo/db/client.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        param: database connection
+    """
+    def __init__(self, connection, *args):
+        self.connection = connection
+
+    def getItem(self, key):
+        """ method for getting items from db  """
+        result = self.connection.get(key)
+
+        if result == 'nil':
```
Comment (rbagrov) [inline week3/emo/db/client.py]: Please check this:
```
In [2]: from db.connection import connection

In [3]: connection.get('pesho')
Out[3]: b"{'password': '121312321321312', 'username': 'pesho'}"

In [4]: connection.get('ddd')

In [5]: connection.get('ddd') is None
Out[5]: True

In [6]: connection.get('ddd') == 'nil'
Out[6]: False

```

**Review by rbagrov — COMMENTED** [review]
Check this out

File: week3/emo/authentication/authenticate.py (hunk 7 above)
Comment (rbagrov) [reply week3/emo/authentication/authenticate.py]: or ```if userExists is None```

**Review by rbagrov — APPROVED** [review]

**Pull request merged**